system/lib/wasmfs/wasmfs.cpp

```diff
@@ -0,0 +1,72 @@
+// // Copyright 2021 The Emscripten Authors.  All rights reserved.
+// // Emscripten is available under two separate licenses, the MIT license and the
+// // University of Illinois/NCSA Open Source License.  Both these licenses can be
+// // found in the LICENSE file.
+// // wasmfs.cpp will implement a new file system that replaces the existing JS filesystem
+// // Current Status: Work in Progress.
+// // See https://github.com/emscripten-core/emscripten/issues/15041
+
+#include <stdlib.h>
+#define __NEED_struct_iovec
+#include <emscripten/emscripten.h>
+#include <wasi/api.h>
+
+extern "C" {
+
+__wasi_errno_t __wasi_fd_write(
+  __wasi_fd_t fd, const __wasi_ciovec_t* iovs, size_t iovs_len, __wasi_size_t* nwritten) {
+  // FD 1 = STDOUT
+  // FD 2 = STDERR
+  // Temporary hardcoding of filedescriptor values
+  // TODO: May not want to proxy stderr (fd == 2) to the main thread
+  // This will not not show in HTML, a console.warn in a worker is suffficient
+  // This would be a change from the current FS
+  if (fd == 1 || fd == 2) {
```

Comment:
Please add `.` at the end of sentences. (We're not 100% consistent there, but let's try in new code :smile: )